pkg/cloud/vsphere/services/govmomi/create.go

```diff
@@ -91,6 +97,23 @@ func generateUserData(ctx *context.MachineContext, bootstrapToken string) ([]byt
 
 	// apply values based on the role of the machine
 	if ctx.HasControlPlaneRole() {
+		if len(ctx.MachineConfig.MachineSpec.Network.Devices) == 0 {
+			return nil, errors.New("machine spec has no network devices")
```

Comment:
Hi @andrewsykim,

Let's move this to the machine actuator and make it a dependency before govmomi create is ever called.

Reply:
sgtm

Reply:
We don't do validation on any field, should we just address this in a follow-up PR?

Reply:
I only added it initially since it's the only place that could cause a panic here

Reply:
> We don't do validation on any field, should we just address this in a follow-up PR?

sgtm